### PR TITLE
chore: add pre-commit

### PR DESCRIPTION
**Pull request by Fabien-GELUS**
Signed-off-by: Fabien Gelus <fabien.gelus@owkin.com>

## Description

Add pre-commit to automatically format and lint before commits. 
Enforce forks to lint their files before the CI is run.

## Ping Reviewers
@melloddy/predictor-users

## Please check if the PR fulfills these requirements

- [ ] If necessary, the [changelog](https://github.com/melloddy/MELLODDY_Predictor/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      name: Format docstrings with `docformatter`
+      language: system
+      types: [python]
+      require_serial: true
+      entry:  docformatter
+      args: [
+        --in-place, --wrap-summaries=88, --wrap-descriptions=87, --recursive
+      ]
+      files: ^(melloddy_predictor|tests|examples)/
+-   repo: local
```
Comment (mblottiere) [inline .pre-commit-config.yaml:14]: There are too many spaces:
```suggestion
- repo: local
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      language: system
+      types: [python]
+      require_serial: true
+      entry:  black
+      args: [
+        --preview,
+        --line-length=120,
+      ]
+      files: ^(melloddy_predictor|tests|examples)/
+-   repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (mblottiere) [inline .pre-commit-config.yaml:36]: Except for this one, all other `repo` entries target `local`, can we group the hooks instead?

File: Makefile [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
 test: inputs
 	pytest ./tests/
 	pytest --nbmake "./examples/"
 	python ./examples/example.py
 
-.PHONY: doc
 doc:
 	pdoc --http localhost:8080 --config show_source_code=False melloddy_predictor
+
+pre-commit-checks: ## Run pre-commit checks on all files
```
Comment (mblottiere) [inline Makefile:16]: `make lint` seems more common and doesn't expose implementation details:
```suggestion
lint: ## Run pre-commit checks on all files
```

**Review by mblottiere — APPROVED** [review]
The example notebook lost some outputs it seems, is it on purpose?

**Comment by Fabien-GELUS** [comment]
> The example notebook lost some outputs it seems, is it on purpose?

it's a good practice to clear output, and automatically done by the `jupyter-nb-clear-output` hook

**Pull request merged**